pkg/controller/generic_reconciler.go

```diff
@@ -307,15 +307,13 @@ func (gr *GenericReconciler) processNamespacedResources(
 	return nil
 }
 
-func (gr *GenericReconciler) reconcileGroupOfObjects(ctx context.Context,
-	objs []*unstructured.Unstructured, namespace string) error {
+func (gr *GenericReconciler) reconcileGroupOfObjects(objs []*unstructured.Unstructured, namespaceUID string) error {
 
 	if gr.allObjectsValidated(objs) {
 		gr.logger.Info("reconcileGroupOfObjects", "All objects are validated", "Nothing to do")
 		return nil
 	}
 
-	namespaceUID := gr.watchNamespaces.getNamespaceUID(namespace)
```

Comment:
what is the advantage of passing the `namespaceUID` instead of getting it here please?

Reply:
getting the namespaceID from here, gets the current namespace's ID. If the namespace was already recreated, you will have the new namespaceID for the old namespace's validated objects.